runtime/lua/vim/lsp/buf.lua

```diff
@@ -827,11 +827,19 @@ local function on_code_action_results(results, opts)
     return
   end
 
-  ---@param item {action: lsp.Command|lsp.CodeAction}
+  ---@param item {action: lsp.Command|lsp.CodeAction, ctx: lsp.HandlerContext}
   local function format_item(item)
-    local title = item.action.title:gsub('\r\n', '\\r\\n')
-    return title:gsub('\n', '\\n')
+    local clients = vim.lsp.get_clients({ bufnr = item.ctx.bufnr })
```

Comment:
`format_item` will be called many times, one for each of the CodeAction entries so this will be inefficient. `clients` can be moved up to the parent scope as a closure.